### ISSUE TITLE
[Bug]: unable to print on Debian 12

### ISSUE THREAD

**Issue by giuliopaci** [electron/electron#38402]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

24.1.2, 24.3.1

### What operating system are you using?

Other Linux

### Operating System Version

Debian 12

### What arch are you using?

x64

### Expected Behavior

I expect the print dialog to let me print.

### Actual Behavior

When I try to print a PDF, the printing dialog is shown, listing printers and print to file options.
None of the available options let me print, instead the dialog is closed without any visible errors/message and without printing anything.

### Additional Information

I have compiled the application using electron forge 6.1.1, using make-deb. I have tried the same deb on 2 Linux boxes, both running Debian. In one of them printing works as expected. In the other one (a fresh install of Debian 12, using RC2 installer and default configuration), the app is not printing.
On the same computer, using the same user, I am able to print using other applications.

The app is attempting to print a PDF using print-js.

I am not sure how to provide further information and a reproducible example, since the issue seems to occur only on specific OS. I am also unsure about how to debug and further investigate the issue.

Do you have any recommendations about how to proceed?

**Comment by giuliopaci** [comment]
@jkleinsc I am able to reproduce the issue reliably. I am just not able to reproduce it in a minimal example without sharing the whole application.
I will follow your suggestion about requesting help through other channels as you suggested and eventually come back if I will be able to isolate the issue.

**Comment by giuliopaci** [comment]
Finally I have been able to reproduce the issue and implement a fix within my application.

TLDR: print-js has this issue https://github.com/crabbly/Print.js/issues/572 which affects most browsers and electron as well. The issue was introduced in version 1.5.0, so using 1.4.0 solves the issue.

As I mentioned in my original request, my application is using print-js to print some PDF and the issue seems hard to reproduce, because it depends on several factors.
In particular, with electron, the issue appears when using print-js 1.5.0 or 1.6.0, when printing PDF-A. For some reason sometimes it still print, but most of the time it does not. In Chrome and Firefox the issue does not happen when printing "normally", but it happens only when using the system dialog to print.

I had issues replicating the issue, because I initially used CDN version of print-js, which is not affected and a different PDF, which is also not affected (or better, which makes the issue happen very rarely).

As far as I understood the issue is due to the fact that print-js load the pdf to be printed into an hidden element and then requests printing. Once the printing dialog is closed, the element is immediately removed, so that it is not available when the browser tries to print it, causing the silent failure. The proposed fix is either to not remove the hidden item or to remove the item after a small delay.

If there is a bug in Electron it is to silently fail without warnings, errors or any other visible effect. However this is the behavior of Chromium as well, when printing through system dialog.
